### ISSUE TITLE
Feature Request: Generate default config file

### ISSUE THREAD

**Issue by issmirnov** [sharkdp/bat#870]
Branching off from #573 - it would be very useful if bat automatically generated a sane, small, well commented config file for users.

For example:
- `bat --config-file -o $file` writes to `$file` 
- `bat --generate-config-file` writes to the current default config location and prints the path

I assume the implementation would be a small static string or binary blog baked into the binary. Not sure how rust does these things.

**Comment by jyn514** [comment]
I don't think the `-o` option is necessary here, you can already redirect the output with `>` at the command line. The other options look useful though.

**Comment by eth-p** [comment]
> I assume the implementation would be a small static string or binary blog baked into the binary. Not sure how rust does these things.

[include_str!](https://doc.rust-lang.org/std/macro.include_str.html) or [include_bytes!](https://doc.rust-lang.org/std/macro.include_bytes.html)

**Comment by sharkdp** [comment]
> I don't think the `-o` option is necessary here, you can already redirect the output with `>` at the command line. The other options look useful though.

Good point, but then how should the CLI really look like?

**Comment by jyn514** [comment]
I would expect `--config-file` to always write to standard out.  Does that answer your question?

**Comment by sharkdp** [comment]
Not really.

```
bat --config-file
```
currently prints the path to the config file, and we are not planning to change that.

The idea of this ticket is to create a new way to generate a default config file, ideally at the place where `bat` would look for it anyway.

My question is how we want to design the CLI to allow users to do that.

If we are implementing something like `bat --print-default-config-file` which prints to the console, are we advising users to run something like this?
```
bat --print-default-config-file > bat --config-file
```
That wouldn't really work because the `~/.config/bat` folder would likely not exist.

**Comment by issmirnov** [comment]
Agree with @sharkdp above - that seems pretty clunky. My vote is on this:

```
$ bat --generate-config-file
Success! Config file written to ~/.config/bat/bat.conf
```

Under the hood, this command creates the `~/.config/bat` folder and writes the config file.

**Comment by sharkdp** [comment]
Closed in #885 by @jmick414

**Comment by issmirnov** [comment]
Woohoo! You guys rock. Thanks @jmick414 and @sharkdp .

**Comment by sharkdp** [comment]
released in [bat v0.14](https://github.com/sharkdp/bat/releases/tag/v0.14.0)

**Comment by GitMurf** [comment]
Is it expected that on Windows the `--generate-config-file` creates a file `C:\Users\USER_NAME\AppData\Roaming\bat\conf` with the config file being named `config` without any extension? This was very confusing as I initially expected I should create a `config` folder and then have the config file go within it naming it something like `config.conf` or `bat.conf` due to this line in the [README](https://github.com/sharkdp/bat#configuration-file): `export BAT_CONFIG_PATH="/path/to/bat.conf"` which references a config file with a `.conf` extension.

The problem that occurred that took me a long time to troubleshoot is that since I had first created a folder `config` when I tried to run the `bat --generate-config-file` I kept getting a PowerShell error that access was denied. It wasn't until I deleted my `config` folder that then the command worked and it generated a config file as `config` without a file extension (as detailed above).

Thoughts? Is this expected?
